src/main/antlr/LibSLLexer.g4

```diff
@@ -272,19 +309,42 @@ CHARACTER
    |   '\'' EscapeSequence '\''
    ;
 
-fragment
-SingleCharacter
+fragment SingleCharacter
     :   ~['\\\r\n]
     ;
 
-fragment
-EscapeSequence
-    :   '\\u' Hex Hex Hex Hex Hex Hex Hex Hex
+fragment EscapeSequence
+    :   UnicodeEscape
+        | OctalEscape
     ;
 
+fragment UnicodeEscape: '\\' 'u'+ Hex Hex Hex Hex;
+
+fragment OctalEscape:
+    '\\' OctalDigit
+    | '\\' OctalDigit OctalDigit
+    | '\\' ZeroToThree OctalDigit OctalDigit
+;
+
+fragment ZeroToThree: [0-3];
+
+fragment Digits: Digit+;
+
 Digit: ('0'..'9');
 
-Hex: Digit | ('A'..'F');
+fragment NonZeroDigit: [1-9];
+
+fragment Hex: Digit | ('A'..'F');
+
+fragment HexNumeral: '0' [xX] Hex+;
```

Comment:
Since the following prefix is expected to be at the beginning of a string, I consider forcing the user to write `0x` (lowercase)
```suggestion
fragment HexNumeral: '0x' Hex+;
```

..or convert to a lowercase/uppercase representation during parsing.